ip: Add support of custom metric for auto routes

cathay4t opened the pull request:
Example:

```yml
---
interfaces:
  - name: eth1
    state: up
    ipv4:
      enabled: true
      dhcp: true
      auto-route-metric: 101
    ipv6:
      enabled: true
      dhcp: true
      autoconf: true
      auto-route-metric: 102
```

TODO: test

ffmancera commented on tests/integration/dynamic_ip_test.py:
```diff
@@ -1388,3 +1372,40 @@ def test_wait_ip(eth1_up, wait_ip):
+        f"ip -{family} -j route show proto dhcp".split(), check=True
+    )[1]
+    ip_routes = json.loads(output)
+    print(ip_routes)
```
I think this is a leftover. The test should not do a print. Thanks!

ffmancera reviewed (changes requested):
Just a nit, other than that, LGTM!

cathay4t committed:
ip: Add support of custom metric for auto routes

Example:

```yml
---
interfaces:
  - name: eth1
    state: up
    ipv4:
      enabled: true
      dhcp: true
      auto-route-metric: 101
    ipv6:
      enabled: true
      dhcp: true
      autoconf: true
      auto-route-metric: 102
```

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
